examples/common/pigweed/rpc_services/Device.h

```diff
@@ -35,6 +36,171 @@
 namespace chip {
 namespace rpc {
 
+namespace Internal {
+// This class supports changing the commissionable data provider values at
+// runtime using the RPCs.
+// This class is LazyInit after using an RPC to read/write any of the data
+// provider values.
+// The class wraps the original CommissionableDataProvider, returning its
+// values for anything which is not overwritten.
+//
+// NOTE: Values written do not persist across a restart.
+class CommissionableDataProviderRpcWrapper : public DeviceLayer::CommissionableDataProvider
+{
+public:
+    CHIP_ERROR GetSetupPasscode(uint32_t & setupPasscode) override
+    {
+        LazyInit();
+        if (mPasscodeOverride.has_value())
+        {
+            setupPasscode = mPasscodeOverride.value();
+            return CHIP_NO_ERROR;
+        }
+        else if (mCommissionableDataProvider)
+        {
+            return mCommissionableDataProvider->GetSetupPasscode(setupPasscode);
+        }
+        return CHIP_ERROR_INTERNAL;
+    }
+
+    CHIP_ERROR SetSetupPasscode(uint32_t setupPasscode) override
+    {
+        LazyInit();
+        mPasscodeOverride = setupPasscode;
+        return CHIP_NO_ERROR;
+    }
```

Comment:
Passcode should be CHIP_ERROR_NOT_IMPLEMENTED since we no longer internally compute the verifier from passcode so it could mismatch